core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java

```diff
@@ -43,110 +53,232 @@
+            {
+                this.type = type;
+                this.pageCount = pageCount;
+            }
+
+            public Page createPage(List<Type> types, int positionCount, float nullRate)
+            {
+                return PageTestUtils.createRandomPage(types, POSITION_COUNT, nullRate);
+            }
+
+            public Type type()
+            {
+                return type;
+            }
+
+            public int pageCount()
+            {
+                return pageCount;
+            }
+        }
+
+        @Setup
+        public void setup()
+        {
+            createPages(type);
+        }
+
+        private void createPages(TestType inputType)
+        {
+            float nullRate = hasNull ? 0.2f : 0.0f;
+
+            types = nCopies(channelCount, inputType.type());
+            dataPage = inputType.createPage(types, POSITION_COUNT, nullRate);
+            pageCount = inputType.pageCount();
+            types = ImmutableList.<Type>builder()
+                    .add(BIGINT) //// dataPage has pre-computed hash block at channel 0
+                    .addAll(types)
+                    .build();
+        }
+
+        private PartitionedOutputBuffer createPartitionedOutputBuffer()
         {
-            BlockTypeOperators blockTypeOperators = new BlockTypeOperators(new TypeOperators());
-            PartitionFunction partitionFunction = new LocalPartitionGenerator(
-                    new InterpretedHashGenerator(ImmutableList.of(BIGINT), new int[] {0}, blockTypeOperators),
-                    PARTITION_COUNT);
-            PagesSerdeFactory serdeFactory = new PagesSerdeFactory(createTestMetadataManager().getBlockEncodingSerde(), false);
             OutputBuffers buffers = createInitialEmptyOutputBuffers(PARTITIONED);
-            for (int partition = 0; partition < PARTITION_COUNT; partition++) {
+            for (int partition = 0; partition < partitionCount; partition++) {
                 buffers = buffers.withBuffer(new OutputBuffers.OutputBufferId(partition), partition);
             }
-            PartitionedOutputBuffer buffer = createPartitionedBuffer(
+
+            return createPartitionedBuffer(
                     buffers.withNoMoreBufferIds(),
-                    DataSize.ofBytes(Long.MAX_VALUE)); // don't let output buffer block
-            TaskContext taskContext = createTaskContext();
-            DriverContext driverContext = createDriverContext(taskContext);
+                    DataSize.of(Long.MAX_VALUE, BYTE));
+        }
+
+        private PartitionedOutputOperator createPartitionedOutputOperator(OperatorFactories operatorFactories, SessionBuilder sessionBuilder)
+        {
+            BlockTypeOperators blockTypeOperators = new BlockTypeOperators(new TypeOperators());
+            PartitionFunction partitionFunction = new LocalPartitionGenerator(
+                    new InterpretedHashGenerator(ImmutableList.of(BIGINT), new int[] {0}, blockTypeOperators),
```

Comment:
since you have pages with hash precomputed, you should use `PrecomputedHashGenerator`